flake.nix

```
@@ -185,7 +185,7 @@
           buildInputs =
             [ rust ]
             ++ missingSysPkgs
-            ++ (if !isDevShell then [ cargoHome ] else [ ]);
+            ++ (if !isDevShell then [ cargoHome ] else [ pkgs.rust-analyzer ]);
```

Minor remark: let's avoid double negative:
```suggestion
            ++ (if isDevShell then [ pkgs.rust-analyzer ] else [ cargoHome ]);
```